golem-cli/src/command_handler/plugin.rs

```diff
@@ -0,0 +1,359 @@
+        Self { ctx }
+    }
+
+    pub async fn handle_command(&self, subcommand: PluginSubcommand) -> anyhow::Result<()> {
+        match subcommand {
+            PluginSubcommand::List { scope } => self.cmd_list(scope).await,
+            PluginSubcommand::Get {
+                plugin_name,
+                version,
+            } => self.cmd_get(plugin_name, version).await,
+            PluginSubcommand::Register { scope, manifest } => {
+                self.cmd_register(scope, manifest).await
+            }
+            PluginSubcommand::Unregister {
+                plugin_name,
+                version,
+            } => self.cmd_unregister(plugin_name, version).await,
+        }
+    }
+
+    async fn cmd_list(&self, scope: PluginScopeArgs) -> anyhow::Result<()> {
+        let (scope_project, scope_component_id) = self.resolve_scope(&scope).await?;
+
+        let plugin_definitions = match self.ctx.golem_clients().await? {
+            GolemClients::Oss(clients) => clients
+                .plugin
+                .list_plugins(Some(&default_plugin_scope(scope_component_id.as_ref())))
+                .await
+                .map(|plugins| {
+                    plugins
+                        .into_iter()
+                        .map(PluginDefinition::from)
+                        .collect::<Vec<_>>()
+                })
+                .map_service_error()?,
+            GolemClients::Cloud(clients) => clients
+                .plugin
+                .list_plugins(Some(&cloud_plugin_scope(
+                    scope_project.as_ref(),
+                    scope_component_id.as_ref(),
+                )))
+                .await
+                .map(|plugins| {
+                    plugins
+                        .into_iter()
+                        .map(PluginDefinition::from)
+                        .collect::<Vec<_>>()
+                })
+                .map_service_error()?,
+        };
+
+        self.ctx.log_handler().log_view(&plugin_definitions);
+
+        Ok(())
+    }
+
+    async fn cmd_get(&self, plugin_name: String, version: String) -> anyhow::Result<()> {
+        let plugin_definition = self.get(&plugin_name, &version).await?;
+        self.ctx.log_handler().log_view(&plugin_definition);
+        Ok(())
+    }
+
+    async fn cmd_register(
+        &self,
+        scope: PluginScopeArgs,
+        manifest: PathBufOrStdin,
+    ) -> anyhow::Result<()> {
+        let (scope_project, scope_component_id) = self.resolve_scope(&scope).await?;
+        let manifest = manifest.read_to_string()?;
+        let manifest: PluginManifest = serde_yaml::from_str(&manifest)
+            .with_context(|| anyhow!("Failed to decode plugin manifest"))?;
+
+        let icon = std::fs::read(&manifest.icon)
+            .with_context(|| anyhow!("Failed to read plugin icon: {}", &manifest.icon.display()))?;
+
+        let specs = match &manifest.specs {
+            PluginTypeSpecificManifest::ComponentTransformer(spec) => {
```

Comment:
This is missing Library and App plugins if I'm not mistaken

Reply:
That is not available for creation yet in the client

Reply:
We should get back to this during the "manifestization" of plugins

Reply:
It should be the in the client, just under different routes:

https://github.com/golemcloud/golem/blob/main/openapi/golem-service.yaml#L3610

Reply:
But i guess there were no commands for it in cli, or am i mistaken? Definitely we should add them, but i was focusing on porting all the exisiting things. If they are straightforward, we can add them, but i would like mostly work on getting all of this into the app manifest, so all of these will become part of deploy.

Reply:
They were not in the old CLI but important to add for the release. The library plugin type is a major simplification we wanted to push in this milestone.

Reply:
Okay, i will add it in the next batch of cleanups.